src/client/src/components/pages/participants/prisonDetails/index.jsx

```diff
@@ -26,17 +29,55 @@ export default class index extends Component {
     this.setState(fields);
   };
 
+
+
+  addPrisonDetails = async newPrisonDetails => {
+    const { id } = this.props.match.params;
+    const confirm = await swal({
+      type: "warning",
+      html: "Are you sure for adding this prison details ?",
+      showCancelButton: true,
+      focusConfirm: false,
+      confirmButtonText: '<i class="fa fa-thumbs-up"></i> Yes',
+      confirmButtonAriaLabel: "Thumbs up",
+      cancelButtonText: '<i class="fa fa-thumbs-down"></i> No ',
+      cancelButtonAriaLabel: "Thumbs down"
+    });
+    if (confirm.value) {
+      const result = await axios(`/api/v2/participants/${id}/prison`, {
+        method: "POST",
+        data: {
+          prisonDetails: newPrisonDetails,
+        }
+      });
+      if (result.data.error) {
+        await swal({
+          title: "",
+          type: "warning",
+          html: result.data.error,
+          confirmButtonText: "Ok"
+        });
+        this.props.history.push("/participants/view");
+      } else {
+        await swal({
+          title: "Success",
+          type: "success",
+          html: result.data.message
+        });
+        this.setState({ ...newPrisonDetails });
```

Comment:
I think you don't need this setState(), since this.state is already equal to newPrisonDetails (i.e. you didn't make any changes to newPrisonDetails in this function).

It's not a required change, but the extra setState call will cause React to re-render your component, even if there are no actual changes to this.state, so it's an extra render that you really don't need.

Reply:
@tomduggan85  :+1: 
you are extremely right :ok_hand: , maybe I will need it if I view the details after the end of adding it. 
but I don't do it so, there is no need to re-rendering  the state.